up-streamer/src/ustreamer.rs

```diff
@@ -420,22 +454,61 @@ impl UStreamer {
     /// * name - Used to uniquely identify this UStreamer in logs
     /// * message_queue_size - Determines size of channel used to communicate between `ForwardingListener`
     ///                        and the worker tasks for each currently endpointd `UTransport`
-    pub fn new(name: &str, message_queue_size: u16) -> Self {
+    /// * usubscription - Subscription service which will be used to store subscription info for topics.
+    pub fn new(name: &str, message_queue_size: u16, usubscription: Arc<dyn USubscription>) -> Self {
         let name = format!("{USTREAMER_TAG}:{name}:");
         // Try to initiate logging.
         // Required in case of dynamic lib, otherwise no logs.
         // But cannot be done twice in case of static link.
+        std::env::set_var("RUST_LOG", "debug");
         let _ = env_logger::try_init();
         debug!(
             "{}:{}:{} UStreamer created",
             &name, USTREAMER_TAG, USTREAMER_FN_NEW_TAG
         );
 
+        let uuri: UUri = UUri {
+            authority_name: "*".to_string(),
+            ue_id: 0x0000_FFFF,     // any instance, any service
+            ue_version_major: 0xFF, // any
+            resource_id: 0xFFFF,    // any
+            ..Default::default()
+        };
+
+        let details_vector = Vec::new();
+        let subscriber_info = SubscriberInfo {
+            uri: Some(uuri).into(),
+            details: details_vector,
+            ..Default::default()
+        };
+
+        let mut fetch_request = FetchSubscriptionsRequest {
+            request: None,
+            offset: None,
+            ..Default::default()
+        };
+        fetch_request.set_subscriber(subscriber_info);
+        let subscriptions = task::block_on(usubscription.fetch_subscriptions(fetch_request));
+        //let subscription_cache_foo = Arc::new(Mutex::new(SubscriptionCache::new(subscriptions.expect("Subscriptions is an Error"))));
+
+        let subscription_cache_foo = match subscriptions {
+            Ok(subs) => Arc::new(Mutex::new(SubscriptionCache::new(subs))),
+            Err(_subs) => {
```

Comment:
I think this is burying the error here.

We should either return an explicit error or panic. I'd lean toward an error of `UStatus` with `UCode::INVALID_ARGUMENT`.